packages/@aws-cdk/core/lib/private/resolve.ts

```diff
@@ -302,8 +303,8 @@ export class RememberingTokenResolver extends DefaultTokenResolver {
  * Not in 'construct.ts' because that would lead to a dependency cycle via 'uniqueid.ts',
  * and this is a best-effort protection against a common programming mistake anyway.
  */
-function isConstruct(x: any): boolean {
-  return x._children !== undefined && x._metadata !== undefined;
+function isClassInstance(x: any): boolean {
```

Comment:
`isClassInstance([]) === true`. Is this ok?

Reply:
Good catch. It actually doesn't matter for correctness of the module, but it might be unexpected.